File: CivitasRuby/pkg/CivitasRuby-0.0.1/lib/diario.rb
module Civitas
  class Diario  
    #esta clase sigue el patrón singleton (única instancia)
    
    private
    def initialize
      @eventos = Array.new
    end
    
    @@instance = Diario.new 
    
    public
    def self.instance
      return @@instance
    end

    def ocurre_evento(e)
      @eventos.push(e)
    end

    def eventos_pendientes
      return (@eventos.length > 0)
    end

    def leer_evento
      e = @eventos.shift
      return e
    end


    
  end
end

File: CivitasJava/src/civitas/Diario.java
package civitas;




import java.util.ArrayList;

public class Diario {
    
  private static final  Diario instance = new Diario();
  
  private ArrayList<String> eventos;
  
  public static Diario getInstance() {
    return instance;
  }
  
  private Diario () {
    eventos = new ArrayList<>();
  }
  
  void ocurreEvento (String e) {
    eventos.add (e);
  }
  
  public boolean eventosPendientes () {
    return !eventos.isEmpty();
  }
  
  public String leerEvento () {
    String salida = "";
    if (!eventos.isEmpty()) {
      salida = eventos.remove(0); //Esto almacena en salida el primer Elemento que está en la lista
    }
    return salida;
  }
  
  
}
